examples/solid-contacts/src/components/ContactDetails.tsx

```diff
@@ -0,0 +1,221 @@
+                    <li class="text-light text-lg flex items-center space-x-4">
+                        <EnvelopeIcon class="w-4 h-4" />
+                        <span> {props.contact.email}</span>
+                    </li>
+                    <span class="text-light text-lg flex items-center space-x-4">
+                        <PhoneIcon class="w-4 h-4" />
+                        <span> {props.contact.phone}</span>
+                    </span>
+                </ul>
+                <div class="flex justify-center items-center space-x-6 pt-10">
+                    <Show when={props.contact.email}>
+                        <Button
+                            label="Send Email"
+                            icon
+                            href={`mailto:${props.contact.email}`}
+                        >
+                            <EnvelopeIcon class="w-6 h-6 m-2"></EnvelopeIcon>
+                        </Button>
+                    </Show>
+                    <Show when={props.contact.phone}>
+                        <Button
+                            icon
+                            href={`tel:${props.contact.phone}`}
+                            label="Make Call"
+                        >
+                            <PhoneIcon class="w-6 h-6 m-2"></PhoneIcon>
+                        </Button>
+                    </Show>
+                </div>
+            </div>
+            <div class="flex items-center space-x-10 justify-around coarse:fixed coarse:bottom-0 coarse:left-0 h-16 w-full px-6">
+                <Button
+                    label="Edit Contact"
+                    href={`/contacts/${props.contact.id}/edit`}
+                    class="flex items-center justify-center flex-col space-y-1.5 !rounded-2xl"
+                >
+                    <PencilIcon class="w-6 h-6"></PencilIcon>
+                    <span class="text-xs">Edit</span>
+                </Button>
+                <FavoriteForm>
+                    <input
+                        type="hidden"
+                        name="favorite"
+                        value={isFavorite() ? 'false' : 'true'}
+                    />
+                    <input
+                        type="hidden"
+                        name="contactId"
+                        value={props.contact.id}
+                    />
+                    <Button
+                        label="Favorite Contact"
+                        type="submit"
+                        class="flex items-center justify-center flex-col space-y-1.5 !rounded-2xl"
+                    >
+                        <Dynamic
+                            component={isFavorite() ? StarSolidIcon : StarIcon}
+                            class="w-6 h-6"
+                            classList={{ 'text-yellow-500': isFavorite() }}
+                        ></Dynamic>
+                        <span class="text-xs">Favorite</span>
+                    </Button>
+                </FavoriteForm>
```

Comment:
This is an issue with `createRouteAction`. When a redirect happens on the server after the form post succeeds, `createRouteAction` does a new navigation (therefore a new entry in history) to the redirected page instead of doing a replace navigation. The back button actually works, but it takes you to the same page. If you click it again, it will actually take you back.

I filed the issue below to get this fixed:
https://github.com/solidjs/solid-start/issues/295